### DISCUSSION TITLE
Play multiple livestreams on a single connection (Android SDK)

### DISCUSSION THREAD

**Discussion by Usama99786** [ant-media/Ant-Media-Server#6454] [Q&A]
I am playing multiple live streams in a recyclerview. Like the live streams play in tiktok. The problem that i am facing is when i switch between the livestreams, I have to first stop the previous stream and then start the next one. This way, the connection is disconnected and connection again which takes some time. I want the videos to play instantly like tiktok. 

  webRTCClient = WebRTCClient(this, context)

          webRTCClient!!.setVideoRenderers(
                viewHolder.mBinding.pipViewRenderer,
                viewHolder.mBinding.cameraViewRenderer
            )

            webRTCClient!!.init(SERVER_URL, streamId, webRTCMode, "", context.intent)

            webRTCClient!!.startStream()
            
 Is there any way for me to achieve the above scenario?

**Comment by nocpbxchap** [comment]
just and idea, maybe you can put not as livestreams but yes as video playback, so it will show and will play and when press will go to the live. just an idea. regards.

**Reply by Usama99786** [reply]
@nocpbxchap thanks for replying, Can you please explain a little bit how can that be done?

**Comment by yashtandon113** [comment]
Hi @Usama99786 
As I see it, the developer replied to your query on the support ticket as well. So let me put the developer's response here as well for other's information.


Build a WebRTC client only once.

If you create it multiple times, each time it will recreate the websocket connection, which will take time. The best way is to hold it in a service on Android.

​When you switch the views you can clear remote video renderers of webrtc client.

`webRTCClient.getConfig().remoteVideoRenderers.clear();
`

Track the resource, then add it again. Call play for the new streamId.

It should work this way.

**Reply by Usama99786** [reply]
Hi @yashtandon113, 
Thanks for replying, Yes i did get this solution on the support ticket, but my problem still remains, Firstly, if I use this approach, Multiple livestreams are being played at the same time. For example when I move to the next stream, even though I have cleared the remoteVideoRenderers, its audio is still being played in the background. 

Secondly, when the stream is played for the first time there is still a loading time for 2-3 seconds. I want to avoid this, In apps like tiktok the loading time to play the livestreams is less than 1 second. Is there anything I can do to make it work like this?

**Reply by asharamseervi** [reply]
Hi @Usama99786 Generally low latency swiping is performed with HLS/MPEG-DASH for the experience you discussed in your use-case of less than 1 second. As the videos are on-demand and easy to 'preload' the first few chunks of upcoming 3-5 videos.

While your requested use-case is strictly with WebRTC; all you need to establish connections, and load upcoming live streams in the background, however bandwidth consumption will be the case of concerns, while loading videos in the background, you may mute the audio of the last 2-3 videos and upcoming 2-3 videos, and only unmute with user swipe/actively watching them.

This is just a workaround, or my 2 cents to collaborate with you guys; probably you need to experiment further, leverage background processes and somehow optimize your algorithms to provide a similar experience to TikTok, for live-streaming/WebRTC.

**Reply by Usama99786** [reply]
Hi @asharamseervi, 
Is there a method i can use to mute the previous or upcoming videos. As I have only one webRtcClient object which I use to play the current stream.
